src/openms/include/OpenMS/FORMAT/ParamCTDFile.h

```diff
@@ -0,0 +1,113 @@
+// --------------------------------------------------------------------------
+//                   OpenMS -- Open-Source Mass Spectrometry
+// --------------------------------------------------------------------------
+// Copyright The OpenMS Team -- Eberhard Karls University Tuebingen,
+// ETH Zurich, and Freie Universitaet Berlin 2002-2021.
+//
+// This software is released under a three-clause BSD license:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Ruben Grünberg $
+// $Authors: Ruben Grünberg $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <string>
+#include <OpenMS/DATASTRUCTURES/Param.h>
+
+namespace OpenMS
+{
+
+  /**
+   @brief A struct to pass information about the tool as one parameter
+   */
+  struct ToolInfo
+  {
+    const std::string& version_;
+    const std::string& name_;
+    const std::string& docurl_;
+    const std::string& category_;
+    const std::string& description_;
+    const std::string& openms_doi_;
```

Comment:
hmm maybe leave it as is and just and rename it to doi_ (as seqan might want to use it as well and making it static might not make sense there)

Reply:
The openms_doi is already a static member of the TOPPBase class so I don't see a reason to make it a member of ParamCTDFile.
Moving is not really an option since some of the parameters are passed from const members of TOPPBase. The only way I see is a constructor that takes r-values for only some of the parameters and that would be more confusing then helpful.

Reply:
Ok and how about leaving out the OpenMS citation/doi then and handling it like ANY general citation, i.e. pass it in the list there?

Reply:
It would probably mean we would need to adapt some code for printing the command line help.
Honestly, I want to pass the ball here as I see no real benefit in spending more time from our side.

Reply:
I just meant in the ParamCTDFile. Not in TOPPBase.
Remove openms_doi from ToolInfo, instead do a citaions.push_back(openms_doi) in TOPPBase. Then remove the special handling in the CTD writing code. 5 mins max

Reply:
Ok Ruben can you look at that?